src/assisted_installer_controller/assisted_installer_controller.go

```diff
@@ -454,37 +478,38 @@ func (c controller) PostInstallConfigs(ctx context.Context, wg *sync.WaitGroup)
 
 	errMessage := ""
 	err, data := c.postInstallConfigs(ctx)
-	// context was cancelled, requires usage of WaitForPredicateWithContext
-	// no reason to set error
-	if ctx.Err() != nil {
-		return
-	}
+	// Even if the context was cancelled, the error is sent to the assisted service.
+	// The timeouts will be handled by the service
 	if err != nil {
 		c.log.Error(err)
 		errMessage = err.Error()
-		c.Status.Error()
+		c.status.Error()
 	}
 	success := err == nil
+	if success {
+		c.updateFinalizingProgress(ctx, models.FinalizingStageDone)
```

Comment:
I think i not fully understand here too, if service is managing the process, why don't we have transition for this state in it? I think service should decide Done or not and not controller

Reply:
Finalizing stages are set by controller.  It is an indication what is the controller doing.

Reply:
I still think done can be set by service itself as it has all the needed data for it , don't see a need for it as next step we send sendCompleteInstallation, so maybe we don't need one of them?